src/Identity/IdentityServer/RequestValidators/BaseRequestValidator.cs

```diff
@@ -348,9 +352,12 @@ private async Task<bool> RequireSsoLoginAsync(User user, string grantType)
         }
 
         // Check if user belongs to any organization with an active SSO policy
-        var anySsoPoliciesApplicableToUser = await PolicyService.AnyPoliciesApplicableToUserAsync(
-                                                user.Id, PolicyType.RequireSso, OrganizationUserStatusType.Confirmed);
-        if (anySsoPoliciesApplicableToUser)
+        var ssoRequired = FeatureService.IsEnabled(FeatureFlagKeys.PolicyRequirements) ?
+            (await PolicyRequirementQuery.GetAsync<RequireSsoPolicyRequirement>(user.Id))
+                .SsoRequired
+            : await PolicyService.AnyPoliciesApplicableToUserAsync(
+                user.Id, PolicyType.RequireSso, OrganizationUserStatusType.Confirmed);
+        if (ssoRequired)
```

Comment:
Formatting here needs fixing, per our usual style:
```
var ssoRequired = condition
  ? case1
  : case2;
```

The `?` needs to be brought down a line.